research/flamby/fed_isic2019/mr_mtl_mkmmd/client.py

```diff
@@ -0,0 +1,200 @@
+
+    def get_criterion(self, config: Config) -> _Loss:
+        return BaselineLoss()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument(
+        "--artifact_dir",
+        action="store",
+        type=str,
+        help="Path to save client artifacts such as logs and model checkpoints",
+        required=True,
+    )
+    parser.add_argument(
+        "--dataset_dir",
+        action="store",
+        type=str,
+        help="Path to the preprocessed FedIsic2019 Dataset (ex. path/to/fedisic2019)",
+        required=True,
+    )
+    parser.add_argument(
+        "--run_name",
+        action="store",
+        help="Name of the run, model checkpoints will be saved under a subfolder with this name",
+        required=True,
+    )
+    parser.add_argument(
+        "--server_address",
+        action="store",
+        type=str,
+        help="Server Address for the clients to communicate with the server through",
+        default="0.0.0.0:8080",
+    )
+    parser.add_argument(
+        "--client_number",
+        action="store",
+        type=int,
+        help="Number of the client for dataset loading (should be 0-5 for FedIsic2019)",
+        required=True,
+    )
+    parser.add_argument(
+        "--learning_rate", action="store", type=float, help="Learning rate for local optimization", default=LR
+    )
+    parser.add_argument(
+        "--lam", action="store", type=float, help="Ditto loss weight for local model training", default=0.01
+    )
+    parser.add_argument(
+        "--seed",
+        action="store",
+        type=int,
+        help="Seed for the random number generators across python, torch, and numpy",
+        required=False,
+    )
+    parser.add_argument(
+        "--mu",
+        action="store",
+        type=float,
+        help="Weight for the mkmmd losses",
+        required=False,
+    )
+    parser.add_argument(
+        "--l2",
+        action="store",
+        type=float,
+        help="Weight for the feature l2 norm loss as a regularizer",
+        required=False,
+    )
+    parser.add_argument(
+        "--mkmmd_loss_depth",
+        action="store",
+        type=int,
+        help="Depth of applying the mkmmd loss",
+        required=False,
+        default=1,
+    )
+    parser.add_argument(
+        "--beta_update_interval",
+        action="store",
+        type=int,
+        help="Interval for updating the beta values",
+        required=False,
+        default=20,
+    )
+    args = parser.parse_args()
+
+    DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    log(INFO, f"Device to be used: {DEVICE}")
+    log(INFO, f"Server Address: {args.server_address}")
+    log(INFO, f"Learning Rate: {args.learning_rate}")
+    log(INFO, f"Lambda: {args.lam}")
+    log(INFO, f"Mu: {args.mu}")
+    log(INFO, f"Feature L2 Norm Weight: {args.l2}")
+    log(INFO, f"MKMMD Loss Depth: {args.mkmmd_loss_depth}")
```

Comment:
Super minor, but probably worth reporting the `beta_update_interval` here as well?